src/benchmark/tpcds_benchmark.cpp

```diff
@@ -0,0 +1,146 @@
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <string>
+
+#include "SQLParser.h"
+#include "SQLParserResult.h"
+#include "benchmark_runner.hpp"
+#include "cli_config_parser.hpp"
+#include "cxxopts.hpp"
+#include "file_based_benchmark_item_runner.hpp"
+#include "file_based_table_generator.hpp"
+#include "json.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/node_queue_scheduler.hpp"
+#include "scheduler/topology.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "sqlite_add_indices.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "utils/assert.hpp"
+#include "visualization/lqp_visualizer.hpp"
+#include "visualization/pqp_visualizer.hpp"
+
+using namespace opossum;  // NOLINT
+
+bool data_files_available(std::string table_path);
+
+int main(int argc, char* argv[]) {
+  auto cli_options = opossum::BenchmarkRunner::get_basic_cli_options("TPC-DS Benchmark");
+
+  // clang-format off
+  cli_options.add_options()
+    ("s,scale", "Database scale factor (1 ~ 1GB)", cxxopts::value<int32_t>()->default_value("1"));
+  // clang-format on
+
+  std::shared_ptr<opossum::BenchmarkConfig> config;
+  int32_t scale_factor;
```

Comment:
in general, prefer the `auto scale_factor = int32_t{}` - this makes sure `scale_factor` always has a defined value (`int32_t{}` is `0`) which, among other things, lessens noise when debugging.